src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -273,6 +280,21 @@ public function wrap($value)
     }
 
     /**
+     * Set the morph map for polymorphic relations.
+     * 
+     * @param  array|null $map
+     * @return array
+     */
+    public static function morphMap($map = null)
+    {
+        if (is_null($map)) {
+            return static::$morphMap;
+        }
+
+        static::$morphMap = $map;
```

Comment:
Right... So perhaps making the method merge the array passed in with the current map? Think that would be a good solution?